packages/trader/src/Modules/Trading/Containers/Multiplier/risk-management-dialog.tsx

```diff
@@ -63,15 +73,15 @@ const RiskManagementDialog = observer(({ is_open, onClose, toggleDialog }) => {
         });
     };
 
-    const onChange = e => {
+    const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
         const { name, value } = e.target;
         const new_state = { ...state };
-        new_state[name] = value;
+        new_state[name as 'take_profit' | 'stop_loss' | 'cancellation_duration'] = value;
```

Comment:
```suggestion
        new_state[name as keyof typeof new_state] = value;
```
will this work?)

Reply:
No, it wouldn't :(
Reason: `new_state` was created based on `state` object, which type is ⤵️
```
state: {
    take_profit: string | undefined;
    stop_loss: string | undefined;
    has_take_profit: boolean;
    has_stop_loss: boolean;
    has_cancellation: boolean;
    cancellation_duration: string;
}
```
So, some of the fields are `boolean` and a `value` (from `e.target`) is a `string`. And here comes an error, that `string` is not assignable to that type (which includes `boolean`). That is why I used union type